File: blind75/Tree/Binary_Tree_Level_Order_Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        treeTraverse(root, 0, res);
        return res;
        
    }
    private void treeTraverse(TreeNode root, int level, List<List<Integer>> res) {
        if (root == null) return;
        if (level == res.size()) res.add(level, new ArrayList<Integer>());
        res.get(level).add(root.val);
        treeTraverse(root.left, level+1, res);
        treeTraverse(root.right, level+1, res);
        
    }
}

File: blind75/Tree/Binary_Tree_Level_Order_Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
      vector<vector<int>> res;
      helper(root, res, 0);
      return res;
    }
    void helper(TreeNode* root, vector<vector<int>> &res, int level) {
      if (!root) return;
      if (res.size() == level) res.push_back(vector<int>());
      res[level].push_back(root->val);
      helper(root->left, res, ++level);
      helper(root->right, res, level);
    }
};
